lib/chef/resource/lwrp_base.rb

```diff
@@ -35,32 +35,42 @@ class LWRPBase < Resource
 
       # Evaluates the LWRP resource file and instantiates a new Resource class.
       def self.build_from_file(cookbook_name, filename, run_context)
-        resource_class = nil
-        rname = filename_to_qualified_string(cookbook_name, filename)
-
-        class_name = convert_to_class_name(rname)
-        if Resource.const_defined?(class_name, false)
-          Chef::Log.info("#{class_name} light-weight resource is already initialized -- Skipping loading #{filename}!")
-          Chef::Log.debug("Overriding already defined LWRPs is not supported anymore starting with Chef 12.")
-          resource_class = Resource.const_get(class_name)
-        else
-          resource_class = Class.new(self)
+        resource_name = filename_to_qualified_string(cookbook_name, filename)
+
+        node = run_context ? run_context.node : Chef::Node.new
+        existing_class = Chef::ResourceResolver.new(node, resource_name).resolve
 
-          Chef::Resource.const_set(class_name, resource_class)
-          resource_class.resource_name = rname
-          resource_class.run_context = run_context
-          resource_class.class_from_file(filename)
+        if existing_class
+          Chef::Log.info("Skipping LWRP resource #{filename} from cookbook #{cookbook_name}.  #{existing_class} already defined.")
+          Chef::Log.debug("Overriding resources with LWRPs is not supported anymore starting with Chef 12.")
+          return existing_class
+        end
```

Comment:
Okay, not sure what code I was reading Friday, but I missed this.  Might be why I don't like returns out of the middle of functions....

I think this might work, but it seems a little roundabout going through the ResourceResolver to look up the resource name.   I'm actually a bit surprised that we don't have a bit of inception since the ResourceResolver is depending on all the provides lines getting setup but we don't setup the default provides and slurp in the class_from_file until a few lines later....